examples/wordpress/backend.jsonnet

```
@@ -0,0 +1,229 @@
+local kube = import "lib/kube.libsonnet";
+
+local labels = {
+  tier: "backend",
+};
+
+local master_labels = labels + {
+  component: "master",
+};
+
+local slave_labels = labels + {
+  component: "slave",
+};
+
+{
+  backend_secret: kube.Secret("mariadb") {
+    metadata+: {
+      labels+: labels,
+    },
+    data_+: {
+      "database_name": "webserver_db",
+      "database_user": "webserver_user",
+      "database_password": "webserver_db_password",
+      "replication_user": "replica_user",
+      "replication_password": "replica_password",
+      "root_user": "root_user",
+      "root_password": "root_password"
+  }},
+
+  backend_master_statefulset: kube.StatefulSet("mariadb-master") {
+    metadata+: {
+      labels+: master_labels,
+    },
+    spec+: {
+      template+: {
+        spec+: {
+          securityContext: {
+            runAsUser: 1001,
+            fsGroup: 1001,
+          },
+          containers_+: {
+            default: kube.Container("mariadb") {
+              image: "bitnami/mariadb",
+              ports_+: {
+                mysql: { containerPort: 3306 },
+              },
+              env_+: {
+                "MARIADB_REPLICATION_MODE": "master",
+                "MARIADB_REPLICATION_USER": { secretKeyRef: { name: "mariadb", key: "replication_user" } },
```

What you have is fine, but also consider:
```suggestion
                MARIADB_REPLICATION_USER: kube.SecretKeyRef($.backend_secret, "replication_user"),
```

The `kube.SecretKeyRef` helper asserts that the `key` actually exists within the secret object, using the secret object directly ensures you don't typo the name, and if the Secret is ever removed you will get an immediate jsonnet error telling you that some other part of the file was still referring to it.

(Ditto for the other similar secretKeyRefs)